src/main/java/io/confluent/connect/elasticsearch/AsyncOffsetTracker.java

```diff
@@ -149,15 +161,19 @@ public synchronized void updateOffsets() {
   }
 
   /**
+   * @param client Elasticsearch client
+   * @param currentOffsets current offsets from a task
    * @return offsets to commit
    */
-  public synchronized Map<TopicPartition, OffsetAndMetadata> offsets() {
+  public synchronized Map<TopicPartition, OffsetAndMetadata> offsets(
+      ElasticsearchClient client,
```

Comment:
Same thing here, this is specific to the sync implementation, I'd remove it from the interface.

Reply:
Good idea, removed.